tempesta_fw/t/functional/README.md

```diff
@@ -0,0 +1,197 @@
+That can be done using `ssh-copy-id`.
+
+
+## Run tests
+
+### Configuration
+
+Testing framework has a lot of options described in `tests_config.ini'.
+You can create default tests configuration by calling:
+
+```sh
+$ ./run_tests.py -d
+```
+Or run next command to show an example of configuration:
+```sh
+$ ./run_tests.py -e
+```
+
+There is 4 sections in configuration: `General`, `Client`, `Tempesta`, `Server`.
+
+#### General Section
+
+`General` section describes the options related to testing framework itself.
+
+`verbose`: verbose level of output:
+- `0` - quiet mode, result of each test is shown by symbols. `.` - passed, `F` -
+failed, `u` - unexpected success, `x` - expected failure. `s` - skipped;
+- `1` - Show test names and doc strings;
+- `2` - Show tests names and performance counters;
+- `3` - Full debug output.
+
+`Duration` option controls duration in seconds of each workload test. Use small
+values to obtain results quickly add large for more heavy stress tests. Default
+is `10` seconds.
+
+`ARP`. Sometimes it is useful to reduce Linux network stack load and load
+TempestaFW internals more, e.g. for performance or stress tests. That can be
+achieved by populating ARP/Neighbour tables on each host. In that case all hosts
+except one running testing framework must be in one subnet. See fully
+distributed configuration. Possible options: `True` or `False`. Default is
+`False`.
+
+This group of options can be overridden by command line options, for more
+information run tests with `-h` key.
+```sh
+$ ./run_tests.py -h
+```
+
+#### Client Section
+
+`ip` - IPv4/IPv6 address of the host in test network. Used to run `wrk`,
+`tempesta`, `nginx` and others with right parameters. Default is `127.0.0.1`.
+
+`mac` - hardware address of interface in test network. Used for populating
+ARP/Neighbour tables. Default is `ff:ff:ff:ff:ff:ff`, which means that ARP
+feature is disabled.
+
+`hostname`, `port`, `user` - this options describes "management" interface of
+the host. Testing framework uses this fields to connect to each test node.
+
+
+Options above are common for all hosts (`Client`, `Server` and `Tempesta`) and
+present in each section.
+
+`ab`, `siege` and `wrk` - absolute path to corresponding binaries.
+
+#### Tempesta Section
+
+Options listed in [Client Section](#client-section): `ip`, `mac`, `hostname`,
+`port`, `user` are also applied to this section. Note, that the `user` must be
+able to load kernel modules.
+
+`dir` - Directory with TempestaFW sources. Must be absolute path.
+
+#### Tempesta Section
+
+Options listed in [Client Section](#client-section): `ip`, `mac`, `hostname`,
+`port`, `user` are also applied to this section.
+
+`nginx` - absolute path to corresponding binary.
+
+`resourses` - absolute path to directory with sample web pages. Must be
+reachable by `nginx`.
+
+
+### Run tests
+
+To run all the tests simply run:
+```sh
+$ ./run_tests.py
```

Comment:
Probably we need the file only. Now `run_all_tests.sh` is just broken, so it's better to remove it at all